skupper-router.spec.rpkg

```diff
@@ -0,0 +1,147 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This .spec file uses `rpkg` (https://pagure.io/rpkg-util) to provide
+# pleasant user experience to developers.
+#
+#  Command                  Description
+# `rpkg srpm`              Creates a *.src.rpm file in /tmp/rpkg (exact path is printed)
+# `rpkg local --nocheck`   Builds a *.rpm for your system in /tmp/rpkg (exact path is printed)
+#
+# See `man rpkg` for more commands.
+
+# https://bugzilla.redhat.com/show_bug.cgi?id=1668512#c19
+%undefine __cmake_in_source_build
+
+%global proton_minimum_version 0.34.0
+%global libwebsockets_minimum_version 3.0.1
+%global libnghttp2_minimum_version 1.33.0
+
+Name:          skupper-router
+Version:       {{{ git_dir_version }}}
+Release:       2.0.0%{?dist}
+Summary:       Router for Skupper.io
+License:       ASL 2.0
+URL:           https://skupper.io
+VCS:           {{{ git_dir_vcs }}}
+
+BuildRequires: cmake
+BuildRequires: python3-devel
+BuildRequires: openssl-devel
+BuildRequires: qpid-proton-c-devel >= %{proton_minimum_version}
+BuildRequires: libwebsockets-devel >= %{libwebsockets_minimum_version}
+BuildRequires: libnghttp2-devel >= %{libnghttp2_minimum_version}
+# man pages --help
+BuildRequires: asciidoc
+BuildRequires: python3-qpid-proton >= %{proton_minimum_version}
+
+Source: {{{ git_dir_pack }}}
+
+%description
+A lightweight message router, written in C and built on Qpid Proton, that provides flexible and scalable interconnect backend for Skupper.io Level 7 Virtual Application Network.
+
+%prep
+{{{ git_dir_setup_macro }}}
+
+%build
+%cmake -DUSE_LIBWEBSOCKETS=ON -DUSE_LIBNGHTTP2=ON -DPython_EXECUTABLE=%{python3}
```

Comment:
On CentOS8, this expands to `-DPython_EXECUTABLE=/usr/libexec/platform-python`. I am not sure this is desired. (Instead of `/usr/bin/python3`). The reason I added this is that on my laptop, CMake would detect Python 3.11 beta and try to build against that.

Reply:
It probably is, since the RPM macro set it this way.